airbyte-server/src/main/java/io/airbyte/server/handlers/MigrationHandler.java

```diff
@@ -0,0 +1,216 @@
+      configRepository.listStandardSources()
+          .forEach(config -> {
+            try {
+              tmpConfigRepository.writeStandardSource(config);
+            } catch (IOException | JsonValidationException e) {
+              throw new RuntimeException(e);
+            }
+          });
+      configRepository.listStandardDestinationDefinitions()
+          .forEach(config -> {
+            try {
+              tmpConfigRepository.writeStandardDestinationDefinition(config);
+            } catch (IOException | JsonValidationException e) {
+              throw new RuntimeException(e);
+            }
+          });
+      configRepository.listSourceConnection()
+          .forEach(config -> {
+            try {
+              tmpConfigRepository.writeSourceConnection(config);
+            } catch (IOException | JsonValidationException e) {
+              throw new RuntimeException(e);
+            }
+          });
+      configRepository.listDestinationConnection()
+          .forEach(config -> {
+            try {
+              tmpConfigRepository.writeDestinationConnection(config);
+            } catch (IOException | JsonValidationException e) {
+              throw new RuntimeException(e);
+            }
+          });
+      configRepository.listStandardSyncs()
+          .forEach(config -> {
+            try {
+              tmpConfigRepository.writeStandardSync(config);
+              tmpConfigRepository.writeStandardSchedule(configRepository.getStandardSyncSchedule(config.getConnectionId()));
+            } catch (IOException | JsonValidationException | ConfigNotFoundException e) {
+              throw new RuntimeException(e);
+            }
+          });
+    } catch (IOException | JsonValidationException | ConfigNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void exportAirbyteDatabase(Path tempFolder) {
+    LOGGER.info(String.format("Exporting Airbyte Database to %s", tempFolder));
+    // TODO implement
+  }
+
+  private void exportVersionFile(Path tempFolder) throws IOException {
+    LOGGER.info(String.format("Exporting Airbyte Version to %s", tempFolder));
+    final String currentVersion = configRepository.getAirbyteVersion();
+    final File versionFile = Files.createFile(tempFolder.resolve(VERSION_FILE_NAME)).toFile();
+    FileUtils.writeStringToFile(versionFile, currentVersion, Charset.defaultCharset());
+  }
+
+  private Path createArchive(Path tempFolder) throws IOException {
+    final Path archiveFile = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz");
+    archiveFile.toFile().deleteOnExit();
+    LOGGER.info(String.format("Creating archive file in %s from %s", archiveFile, tempFolder));
+    final TarArchiveOutputStream archive =
+        new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(archiveFile.toFile()))));
+    Files.walkFileTree(tempFolder, new SimpleFileVisitor<>() {
```

Comment:
woof. what a hassle! seems like the agreed upon way of doing it though.

Reply:
I moved it into a archiveHelper class, should I move that into some airbyte-common package or leave it in this module?

Reply:
In #1507, you probably want to uncompress/recompress a tarball (or do you expect the user to do so manually?) so I moved the utility class to common so you can re-use these functions in the airbyte-migration module too